lib/services/project-service.ts

```diff
@@ -50,10 +51,12 @@ export class ProjectService implements IProjectService {
 
 			await this.ensureAppResourcesExist(projectDir);
 
-			let packageName = constants.TNS_CORE_MODULES_NAME;
-			await this.$npm.install(packageName, projectDir, { save: true, "save-exact": true });
-
 			let templatePackageJsonData = this.getDataFromJson(templatePath);
+
+			if (!(templatePackageJsonData && templatePackageJsonData.dependencies && templatePackageJsonData.dependencies[constants.TNS_CORE_MODULES_NAME])) {
```

Comment:
@rosen-vladimirov  Isn't it more correct to install the correct version of the Template? I.e. if you install 3.0.0 template and you override the tns_core_modules, won't still there be a possibility that this 3.0.0 template doesn't work with them? For pure NS I don't know, but for NS+Angular I think it's likely?